api/v1alpha1/envoyfleet_types.go

```diff
@@ -126,6 +132,21 @@ type AccessLoggingConfig struct {
 	JsonTemplate map[string]string `json:"json_template,omitempty"`
 }
 
+type TLS struct {
```

Comment:
We need:
//+optional
for each of these except TlsSecrets.
We also need descriptive comments above for each of these in order for controller-tools to generate descriptions in K8s CRD.